hail/python/hail/expr/expressions/typed_expressions.py

```diff
@@ -3617,8 +3617,12 @@ def __getitem__(self, item):
                         stop = hl.cond(step > 0, dlen, to_expr(-1, tint64))
 
                     slices.append(hl.tuple((start, stop, step)))
-                else:
-                    slices.append(s)
+                else:  # Not a slice, just an int index
```

Comment:
We can avoid this comment; the typechecker already requires this to be either a slice or an int64